dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/TokenRequestor.java

```diff
@@ -99,7 +106,7 @@ private List<NameValuePair> getBasicParams() {
         return params;
     }
 
-    void addPrivateKeyJwtParameters(List<NameValuePair> params) {
+    void addPrivateKeyJwtParameters(List<NameValuePair> params) throws TokenRequestException {
```

Comment:
**code enhancement opportunities for next pr:**

this method can probably be moved to:
dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/common/token/auth/PrivateKeyJwtAuthMethod.java

we can pass in the params via `customParams` in the builder.

`PrivateKeyJwtAuthMethod` will end up being something like

```java
    @Override
    public void setAuthMethodSpecificSettings(Builder tokenRequestBuilder) {
        tokenRequestBuilder.clientAssertionSigningAlgorithm(clientConfig.getTokenEndpointAuthSigningAlgorithm());
        Key clientAssertionSigningKey = getKeyForPrivateKeyJwtClientAssertion();
        tokenRequestBuilder.clientAssertionSigningKey(clientAssertionSigningKey);

        vvvvvvv
        Map<String, String> clientAssertionParams = getClientAssertionParams();
        tokenRequestBuilder.customParams(clientAssertionParams);
        ^^^^^^^
    }
    
    private Map<String, String> getClientAssertionParams() {
        Map<String, String> clientAssertionParams = new HashMap<>();
        clientAssertionParams.add(new BasicNameValuePair(TokenConstants.CLIENT_ASSERTION_TYPE, TokenConstants.CLIENT_ASSERTION_TYPE_JWT_BEARER));
        String clientAssertionJwt = buildClientAssertionJwt();
        clientAssertionParams.add(new BasicNameValuePair(TokenConstants.CLIENT_ASSERTION, clientAssertionJwt));
        return clientAssertionParams;
    }
    ...
```

perhaps the `createPrivateKeyJwt` method from:
dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/auth/PrivateKeyJwtAuthMethod.java can be moved to here as well, so we don't end up with 2 files named `PrivateKeyJwtAuthMethod`.

also, if we create a `PostAuthMethod` and `ClientSecretPostAuthMethod` which both extends from your new `TokenEndpointAuthMethod`, then we can get rid of this chunk entirely:
https://github.com/OpenLiberty/open-liberty/blob/e46a4d1210048ec9bdb2967d84f6783ebcc69068/dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/token/TokenRequestor.java#L100-L105

`PostAuthMethod` and `ClientSecretPostAuthMethod` can both just pass in the `clientId` and `clientSecret` using the same `customParams` method i presented above.